pallets/did/src/lib.rs

```diff
@@ -114,87 +116,177 @@ pub mod pallet {
+				let mut info = maybe_info.as_mut().ok_or(Error::<T>::IssuerDoesNotExist)?;
+				ensure!(
+					*info == IssuerInfo { status: IssuerStatus::Revoked },
+					Error::<T>::IssuerNotRevoked
+				);
+				*info = IssuerInfo { status: IssuerStatus::Active };
+				Self::deposit_event(Event::IssuerStatusReactived { issuer });
+				Ok(())
+			})?;
+			Ok(())
+		}
+
+		#[pallet::call_index(4)]
 		#[pallet::weight(1000000)]
 		pub fn delete_issuer(origin: OriginFor<T>, issuer: DidIdentifierOf<T>) -> DispatchResult {
-			// Called by the user to de
+			// Origin ONLY GovernanceOrigin
 			T::GovernanceOrigin::ensure_origin(origin)?;
-			// Remove issuer from storage
-			// Should be also called when a Issuer DID is deleted()
-			// because of that prob create a do_delete_issuer() pallet helper method
+
+			Issuers::<T>::try_mutate_exists(issuer.clone(), |maybe_info| -> DispatchResult {
+				let mut info = maybe_info.as_mut().ok_or(Error::<T>::IssuerDoesNotExist)?;
+				ensure!(
+					*info == IssuerInfo { status: IssuerStatus::Revoked },
+					Error::<T>::IssuerNotRevoked
+				);
+				// Remove issuer from storage
+				*maybe_info = None;
+				Self::deposit_event(Event::IssuerDeleted { issuer });
+				Ok(())
+			})?;
 			Ok(())
 		}
 
+		#[pallet::call_index(5)]
 		#[pallet::weight(1000000)]
-		pub fn add_credential_type(origin: OriginFor<T>, credential: CredentialOf<T>) -> DispatchResult {
- 			// Origin ONLY GovernanceOrigin
+		pub fn add_credential_type(
+			origin: OriginFor<T>,
+			credential: CredentialOf<T>,
+		) -> DispatchResult {
+			// Origin ONLY GovernanceOrigin
 			T::GovernanceOrigin::ensure_origin(origin)?;
+			let mut credentials_types = CredentialsTypes::<T>::get();
+			let pos = credentials_types
+				.binary_search(&credential)
+				.err()
+				.ok_or(Error::<T>::CredentialAlreadyAdded)?;
+			credentials_types
+				.try_insert(pos, credential.clone())
+				.map_err(|_| Error::<T>::MaxCredentials)?;
+
+			CredentialsTypes::<T>::put(credentials_types);
+			Self::deposit_event(Event::CredentialTypeAdded { credential });
 			Ok(())
 		}
 
+		#[pallet::call_index(6)]
 		#[pallet::weight(1000000)]
-		pub fn remove_credential_type(origin: OriginFor<T>, credential: CredentialOf<T>) -> DispatchResult {
+		pub fn remove_credential_type(
+			origin: OriginFor<T>,
+			credential: CredentialOf<T>,
+		) -> DispatchResult {
 			// Origin ONLY GovernanceOrigin
 			T::GovernanceOrigin::ensure_origin(origin)?;
+			let mut credentials_types = CredentialsTypes::<T>::get();
+			let pos = credentials_types
+				.binary_search(&credential)
+				.ok()
+				.ok_or(Error::<T>::CredentialDoesNotExist)?;
+			credentials_types.remove(pos);
+			CredentialsTypes::<T>::put(credentials_types);
+			Self::deposit_event(Event::CredentialTypeRemoved { credential });
+			Ok(())
+		}
+	}
+
+	impl<T: Config> Pallet<T> {
+		pub fn do_delete_issuer(origin: OriginFor<T>, issuer: DidIdentifierOf<T>) -> DispatchResult {
```

Comment:
```suggestion
		pub fn do_remove_issuer(origin: OriginFor<T>, issuer: DidIdentifierOf<T>) -> DispatchResult {
```
Just to be consistent with the rest of extrinsics naming